typescript/infra/scripts/agents/update-agent-config.ts

```diff
@@ -138,10 +144,18 @@ export async function writeAgentConfig(
     additionalConfig,
   );
 
-  writeMergedJSONAtPath(
-    getAgentConfigJsonPath(envNameToAgentEnv[environment]),
-    agentConfig,
-  );
+  const filepath = getAgentConfigJsonPath(envNameToAgentEnv[environment]);
+  if (fs.existsSync(filepath)) {
+    const currentAgentConfig: AgentConfig = readJSONAtPath(filepath);
+    // Remove transactionOverrides from each chain in the agent config
+    // To ensure all overrides are configured in infra code or the registry, and not in JSON
+    for (const chainConfig of Object.values(currentAgentConfig.chains)) {
+      delete chainConfig.transactionOverrides;
+    }
+    writeJsonAtPath(filepath, objMerge(currentAgentConfig, agentConfig));
+  } else {
+    writeJsonAtPath(filepath, agentConfig);
+  }
```

Comment:
currently tx overrides never get removed - only added or updated. this change ensures that we correctly derive these values from chains.ts or from the registry